tests/components/plugwise/test_config_flow.py

```diff
@@ -435,3 +436,82 @@ async def test_zeroconf_abort_anna_with_adam(hass: HomeAssistant) -> None:
     flows_in_progress = hass.config_entries.flow._handler_progress_index[DOMAIN]
     assert len(flows_in_progress) == 1
     assert list(flows_in_progress)[0].product == "smile_open_therm"
+
+
+async def _start_reconfigure_flow(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    host_ip: str,
+) -> ConfigFlowResult:
+    """Initialize a reconfigure flow."""
+    mock_config_entry.add_to_hass(hass)
+
+    reconfigure_result = await mock_config_entry.start_reconfigure_flow(hass)
+
+    assert reconfigure_result["type"] is FlowResultType.FORM
+    assert reconfigure_result["step_id"] == "reconfigure"
+
+    return await hass.config_entries.flow.async_configure(
+        reconfigure_result["flow_id"], {CONF_HOST: host_ip}
+    )
+
+
+async def test_reconfigure_flow(
+    hass: HomeAssistant,
+    mock_smile_adam: AsyncMock,
+    mock_setup_entry: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test reconfigure flow."""
+    result = await _start_reconfigure_flow(hass, mock_config_entry, TEST_HOST)
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "reconfigure_successful"
+
+    entry = hass.config_entries.async_get_entry(mock_config_entry.entry_id)
+    assert entry
+    assert entry.data.get(CONF_HOST) == TEST_HOST
+
+
+async def test_reconfigure_flow_other_smile(
+    hass: HomeAssistant,
+    mock_smile_adam: AsyncMock,
+    mock_setup_entry: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test reconfigure flow aborts on other Smile ID."""
+    mock_smile_adam.smile_hostname = TEST_SMILE_HOST
+
+    result = await _start_reconfigure_flow(hass, mock_config_entry, TEST_HOST)
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "not_the_same_smile"
+
+
+@pytest.mark.parametrize(
+    ("side_effect", "reason"),
+    [
+        (ConnectionFailedError, "cannot_connect"),
+        (InvalidAuthentication, "invalid_auth"),
+        (InvalidSetupError, "invalid_setup"),
+        (InvalidXMLError, "response_error"),
+        (RuntimeError, "unknown"),
+        (UnsupportedDeviceError, "unsupported"),
+    ],
+)
+async def test_reconfigure_flow_errors(
```

Comment:
```suggestion
async def test_reconfigure_flow_connect_errors(
```
all errors are from the `connect()` call, right?

Reply:
Yah, this is mostly copying from the existing user flow.